### PR TITLE
Add support for Razer Ornata V3 [028F]

### LINKED ISSUE #1925: Support for Razer Ornata V3 [028F]
### Device Info

* Name: Razer Ornata v3
* Product Number: RZ03-04460300-R3W1
* Model Number: RZ03-0446

### Device Information
```
Please post the output of
    lsusb -d 1532: -v

Bus 001 Device 029: ID 1532:028f Razer USA, Ltd Razer Ornata V3
Couldn't open device, some information will be missing
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.00
  bDeviceClass            0 
  bDeviceSubClass         0 
  bDeviceProtocol         0 
  bMaxPacketSize0        64
  idVendor           0x1532 Razer USA, Ltd
  idProduct          0x028f 
  bcdDevice            2.00
  iManufacturer           1 Razer
  iProduct                2 Razer Ornata V3
  iSerial                 0 
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength       0x0054
    bNumInterfaces          3
    bConfigurationValue     1
    iConfiguration          0 
    bmAttributes         0xa0
      (Bus Powered)
      Remote Wakeup
    MaxPower              500mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      1 Boot Interface Subclass
      bInterfaceProtocol      1 Keyboard
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      61
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x81  EP 1 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0008  1x 8 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        1
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      0 
      bInterfaceProtocol      1 Keyboard
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength     179
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x82  EP 2 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0016  1x 22 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        2
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      0 
      bInterfaceProtocol      2 Mouse
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      94
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x83  EP 3 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0008  1x 8 bytes
        bInterval               1

```

```
Please post the output of
    ls -lah /dev/input/by-id/

lrwxrwxrwx 1 root root  10 Oct 11 16:39 usb-Razer_Razer_Ornata_V3-event-if01 -> ../event20
lrwxrwxrwx 1 root root   9 Oct 11 16:39 usb-Razer_Razer_Ornata_V3-event-kbd -> ../event2
lrwxrwxrwx 1 root root   9 Oct 11 16:39 usb-Razer_Razer_Ornata_V3-if01-event-kbd -> ../event4
lrwxrwxrwx 1 root root  10 Oct 11 16:39 usb-Razer_Razer_Ornata_V3-if02-event-mouse -> ../event21
lrwxrwxrwx 1 root root   9 Oct 11 16:39 usb-Razer_Razer_Ornata_V3-if02-mouse -> ../mouse0

```

### Packet Captures

<Please see here as we probably need packet captures>
<https://github.com/openrazer/openrazer/wiki/Reverse-Engineering-USB-Protocol>

### Synapse Screenshots
<Screenshots of all available options as seen on **Razer Synapse** in a Windows Environment>

### LINKED ISSUE #2309: Support for razer ornata v3 028f (arch)
### Affected Device / PID

Razer Ornata V3, 1532:028f

polychromatic doesn't see my keyboard at all, I tried to find any solution on github, none of them worked
Screenshot_23-Sep_17-21-36_19138

Name: Razer Ornata v3
Product Number: RZ03-04460100-R3M1
Model Number: RZ03-0446

[OrnataV3_information_device.txt](https://github.com/user-attachments/files/17101275/OrnataV3_information_device.txt)

### PR DESCRIPTION
**Pull request by tulser**
This is to add support for the `028f` PID variant of the Razer Ornata V3, following the work from #2182.

#2182 originally contained support for both `02a1` and `028f` PID variants. While the former variant was reported to work without issues by different users, one user, @ddB0515, [reported issues and errors](https://github.com/openrazer/openrazer/pull/2182#issuecomment-1933990284) with the latter despite being able to set chroma. Support for the `02a1` variant was merged instead, but these issues with `028f` have yet to be narrowed down and no other users have entered to admit possessing this keyboard variant in the related threads.

Fixes #1925.

### CODE REVIEW

File: scripts/generate_fake_driver.sh
Comment (z3ntu) [inline scripts/generate_fake_driver.sh]: I think if you do 028F here the _fake_driver diff should be cleaner. But doesn't matter much also

**Comment by z3ntu** [comment]
Code-wise looks good. So now it needs definitely some testing, and then potentially some fixing of whatever issues are found. Thanks!

**Comment by RobQnadqs** [comment]
I've been using this for about a week now, and haven't found any issues so far. I'm using it with Polychromatic, and tested most of the features they offer:
- adjusting brightness
- applying an RGB effect
- enabling Game Mode
- creating and playing a custom RGB effect

All of this worked without any issue. I will post here if I find any issues in the future. Let me know if there are any features that need testing that I haven't tried yet.

**Comment by tulser** [comment]
Hey @RobQnadqs, good to hear it seems to work for you. Just to verify that it is working fine, could you see if you are having any issues with Polychromatic freezing and errors being emitted from the driver while changing modes as brought up in [this comment?](https://github.com/openrazer/openrazer/pull/2182#issuecomment-1933990284)
You can run `sudo dmesg -w` in a terminal and keep it open/running while changing modes in Polychromatic to check for said errors.

**Comment by RobQnadqs** [comment]
Hi @tulser, I tried the exact same thing as in the video in [this comment](https://github.com/openrazer/openrazer/pull/2182#issuecomment-1935846131), and it worked for me! Polychromatic was not freezing, and there were no error messages in the terminal in which I ran `sudo dmesg -w` as you suggested.

**Comment by tulser** [comment]
@RobQnadqs Awesome to hear. Just a final thing, is your specific keyboard's product number `RZ03-04460300-R3W1` or is it a different product number? You should be able to find the product number on the back label near `P\NO`.
Also, if that product numbers differs, would you mind also posting that product number just so it can be noted?

**Comment by RobQnadqs** [comment]
The product number of my keyboard is different, it's `RZ03-04460100-R3M1`.

**Comment by z3ntu** [comment]
I believe everything after `RZ03-0446` is just some regional layout difference, and at least so far we haven't had any behavior differences between those.

**Comment by tulser** [comment]
> I believe everything after `RZ03-0446` is just some regional layout difference, and at least so far we haven't had any behavior differences between those.

I figured the latter part of the Product No. referred to the layout. I was making sure that if the keyboard numbers were indicated to match, then, in the unlikely event that there was some hardware difference, it would definitively confirm that only ddB0515 has the specific issue.

Now that it does not, it still means the `RZ03-04460300-R3W1` could have issues unless ddB0515 has since been able to identify the problem and it's not related to the driver (of the PR), his compiled driver, or potentially his keyboard itself.

---

@ddB0515: Were you ever able to identify the problem on your end? [I recall asking if you would be able to do a USB pcap while changing modes](https://github.com/openrazer/openrazer/pull/2182#issuecomment-1937106910) and I was wondering if you could get around to post one at some point if you haven't already rectified the problem.

**Comment by z3ntu** [comment]
Any update here? I'm looking to do an OpenRazer release quite soon, but no rush if it's not ready yet. :)

**Comment by tulser** [comment]
> Any update here?

Not much since @ddB0515 hasn't replied to my last post, even going on a week now. I was hoping for him to comment if he's had any further investigation into his issue so that any potential differences relating to his keyboard, rather than some other defect or issue either on his specific host or hardware, could be ruled out.

Since there has been one keyboard that has been confirmed to work with this PR that has none of the issues that were brought up, I personally would be confident to say that this is altogether good to merge despite preferring all involved to confirm their keyboards/drivers functionally and properly work together, but it's your call @z3ntu if you'd like to delay this.

**Comment by ddB0515** [comment]
Hello all :wave: 
Was on holidays and apologies for not answering @tulser, I see a lot happened since last time and I didn't test more after but if you give me a bit of time to get newest version and test what's changed and if there is any issues/problems will report back

**Comment by tulser** [comment]
Okay! Then @z3ntu now that there's been a response, I now think we can hold on merging this for a bit.

**Comment by ddB0515** [comment]
@tulser can confirm that I didn't have any issues or problems, everything working fine 
Switching also works without freezing or any kind of issues so hope now is ready for merge :muscle: 
Thanks a lot

**Comment by tulser** [comment]
Alright, if it's all good, then @z3ntu LGTM if you have no other concerns.

**Review by z3ntu — APPROVED** [review]
LGTM, thanks for the patch!

**Pull request merged**
